lib/pure/oids.nim

```diff
@@ -36,12 +37,8 @@ proc hash*(oid: Oid): Hash =
   h = h !& hash(oid.count)
   result = !$h
 
-proc hexbyte*(hex: char): int =
-  case hex
-  of '0'..'9': result = (ord(hex) - ord('0'))
-  of 'a'..'f': result = (ord(hex) - ord('a') + 10)
-  of 'A'..'F': result = (ord(hex) - ord('A') + 10)
-  else: discard
+proc hexbyte*(hex: char): int {.inline.} =
```

Comment:
this works but is convoluted, and maybe inefficient. instead:
add an overload in std/private/decode_helpers:

```nim
proc handleHexChar*(c: char): int {.inline.} =
  case c
  of '0'..'9': result = (ord(c) - ord('0'))
  of 'a'..'f': result = (ord(c) - ord('a') + 10)
  of 'A'..'F': result = (ord(c) - ord('A') + 10)
  else: discard
```

and use that instead
* more readable / less obfuscated, maybe more efficient
* other procs you previously changed to use `handleHexChar(hex, result)` in other PR's might perhaps use this overload
* the 2 overloads are in the same module `decode_helpers` so the duplication is acceptable; maybe in future they can be merged, or not, TBD